packages/shadydom/src/patches/ParentNode.js

```diff
@@ -93,7 +93,36 @@ export const ParentNodePatches = utils.getOwnPropertyDescriptors({
       return children.length;
     }
     return 0;
-  }
+  },
+
+  /** @this {Element} */
+  append(...args) {
+    for (const arg of args) {
+      const newChild = typeof arg === 'string' ? document.createTextNode(arg) : arg;
+      this[utils.SHADY_PREFIX + 'insertBefore'](newChild);
+    }
+  },
+
+  /** @this {Element} */
+  prepend(...args) {
+    const firstChild = this[utils.SHADY_PREFIX + 'firstChild'];
+    for (const arg of args) {
+      const newChild = typeof arg === 'string' ? document.createTextNode(arg) : arg;
+      this[utils.SHADY_PREFIX + 'insertBefore'](newChild, firstChild);
+    }
+  },
+
+  /** @this {Element} */
+  ['replaceChildren'](...args) {
+    let child;
+    while ((child = this[utils.SHADY_PREFIX + 'firstChild']) !== null) {
+      this[utils.SHADY_PREFIX + 'removeChild'](child);
+    }
+    for (const arg of args) {
+      const newChild = typeof arg === 'string' ? document.createTextNode(arg) : arg;
+      this[utils.SHADY_PREFIX + 'insertBefore'](newChild);
+    }
```

Comment:
I should check if this is the right order (remove existing children, then insert new children), given that it probably feeds into MutationObserver.

Reply:
This looks like the correct order: [`replaceChildren`](https://dom.spec.whatwg.org/#dom-parentnode-replacechildren) calls ['replace all'](https://dom.spec.whatwg.org/#concept-node-replace-all), which removes first and inserts second.